src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/srm/algebras/BV.cs

```diff
@@ -159,59 +156,34 @@ public static BV Mk(params bool[] bits)
             int j = x.Length % 64;
             if (j > 0)
             {
-                //the number of bits is not a precise multiple of 64
-                //so the last block has extra bits that need to be reset to 0
+                // the number of bits is not a precise multiple of 64
+                // reset the extra bits in the last block to 0
                 int last = (x.Length - 1) / 64;
-                blocks[last] &= (UL1 << j) - 1;
+                blocks[last] &= (1ul << j) - 1;
             }
 
             return new BV(x.Length, blocks);
         }
 
-        /// <summary>
-        /// less than
-        /// </summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool operator <(BV x, BV y) => x.CompareTo(y) < 0;
-
-        /// <summary>
-        /// greater than
-        /// </summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool operator >(BV x, BV y) => x.CompareTo(y) > 0;
-
-        /// <summary>
-        /// less than or equal
-        /// </summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool operator <=(BV x, BV y) => x.CompareTo(y) <= 0;
-
-        /// <summary>
-        /// greater than or equal
-        /// </summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool operator >=(BV x, BV y) => x.CompareTo(y) >= 0;
-
         /// <summary>
         /// Returns the serialized representation
         /// </summary>
         public override string ToString() => SerializeToString();
 
         public override bool Equals(object obj) => CompareTo(obj) == 0;
 
-        private int _hashcode;
         public override int GetHashCode()
         {
-            if (_hashcode == 0)
+            // if the hash code hasn't been calculated yet, do so before returning it
+            if (_hashcode == null)
             {
                 _hashcode = Length.GetHashCode();
                 for (int i = 0; i < _blocks.Length; i++)
                 {
-                    _hashcode = (_hashcode << 1) ^ _blocks[i].GetHashCode();
+                    _hashcode = HashCode.Combine(_hashcode, _blocks[i].GetHashCode());
                 }
```

Comment:
Could this be:
```C#
HashCode c = default;
c.Add(Length);
c.AddBytes(MemoryMarshal.AsBytes(_blocks));
```
?

Reply:
Yes, this looks cleaner and more efficient. If I understand it correctly, the point here is to calculate a hashcode for a 
sequence (_blocks) of ulongs (typical case being _blocks.Length = 2 or at most 3).

Reply:
Just a note, if we replace BV with BitVector we still need a separate hash code implementation since BitVector doesn't implement one.

Reply:
I think so, yes.